src/Common/AsynchronousMetrics.cpp

```diff
@@ -68,6 +68,8 @@ AsynchronousMetrics::AsynchronousMetrics(
 
     openFileIfExists("/sys/fs/cgroup/memory/memory.limit_in_bytes", cgroupmem_limit_in_bytes);
     openFileIfExists("/sys/fs/cgroup/memory/memory.usage_in_bytes", cgroupmem_usage_in_bytes);
+    openFileIfExists("/sys/fs/cgroup/memory/memory.max", cgroupmem_v2_limit_in_bytes);
+    openFileIfExists("/sys/fs/cgroup/memory/memory.current", cgroupmem_v2_usage_in_bytes);
```

Comment:
I think its nested one level too deep and should be `/sys/fs/cgroup/memory.max`;

```bash
root@1e60725e86d2:/# cd /sys/fs/cgroup
root@1e60725e86d2:/sys/fs/cgroup# ls -la
total 0
drwxr-xr-x 2 root root 0 Feb  3 09:35 .
drwxr-xr-x 8 root root 0 Feb  3 09:35 ..
-r--r--r-- 1 root root 0 Feb  3 09:35 cgroup.controllers
-r--r--r-- 1 root root 0 Feb  3 09:35 cgroup.events
...
-rw-r--r-- 1 root root 0 Feb  3 09:35 cpu.idle
-rw-r--r-- 1 root root 0 Feb  3 09:35 cpu.max
-rw-r--r-- 1 root root 0 Feb  3 09:36 cpu.max.burst
-rw-r--r-- 1 root root 0 Feb  3 09:36 cpu.pressure
-r--r--r-- 1 root root 0 Feb  3 09:35 cpu.stat
-rw-r--r-- 1 root root 0 Feb  3 09:35 cpu.weight
-rw-r--r-- 1 root root 0 Feb  3 09:36 cpu.weight.nice
-rw-r--r-- 1 root root 0 Feb  3 09:35 cpuset.cpus
...
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.current
-r--r--r-- 1 root root 0 Feb  3 09:35 memory.events
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.events.local
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.high
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.low
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.max
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.min
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.numa_stat
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.oom.group
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.peak
-rw-r--r-- 1 root root 0 Feb  3 09:36 memory.pressure
--w------- 1 root root 0 Feb  3 09:36 memory.reclaim
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.stat
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.swap.current
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.swap.events
-rw-r--r-- 1 root root 0 Feb  3 09:36 memory.swap.high
-rw-r--r-- 1 root root 0 Feb  3 09:35 memory.swap.max
-r--r--r-- 1 root root 0 Feb  3 09:36 memory.zswap.current
-rw-r--r-- 1 root root 0 Feb  3 09:36 memory.zswap.max
...
root@1e60725e86d2:/sys/fs/cgroup# 
root@1e60725e86d2:/sys/fs/cgroup# cat memory.current 
8265728
root@1e60725e86d2:/sys/fs/cgroup# 
root@1e60725e86d2:/sys/fs/cgroup# cat memory.max     
8388608
```

Reply:
And note that when NO limit is specified for a container, the file will read `max`

```bash
root@55601c494090:/# cat /sys/fs/cgroup/memory.max 
max
```

Reply:
What I found also is that sometimes those memory files get displaced, likely depending on OS / systemd / runc / containerd / docker versions etc.

For example Systemd v232+ uses
`/sys/fs/cgroup/memory<memory_cgroup_identifier>/memory.limit_in_bytes`

cgroupv1 might use
`/sys/fs/cgroup/memory/<cgroup_identifier>/memory.limit_in_bytes`

cgroupv2 might use
`/sys/fs/cgroup/<cgroup_identifier>/memory.max`

Reply:
updated for too deep path. systemd v232+ with cgroup id may be a bit more complex logic, let me think about it and add follow up PR for it.

Reply:
The `max` value would also break in current implementation? Looks like it would `readText` correctly but then try to insert a string in a `int` column.

Reply:
i think now at least max should be treated